src/rpm_ostree/cli_status.rs

```diff
@@ -143,24 +144,23 @@ fn booted_json(status: &StatusJSON) -> Fallible<DeploymentJSON> {
     Ok(booted)
 }
 
-/// Introspect deployments (rpm-ostree status) using rpm-ostree client actor client's
-/// cache if possible.
-fn status_json(client: &mut RpmOstreeClient) -> Fallible<StatusJSON> {
+/// Ensure our status cache is up to date; if empty or out of date, run `rpm-ostree status` to populate it.
+fn status_json(client: &mut RpmOstreeClient) -> Fallible<Rc<StatusJSON>> {
     STATUS_CACHE_ATTEMPTS.inc();
     let ostree_depls_data = fs::metadata(OSTREE_DEPLS_PATH)
         .with_context(|e| format_err!("failed to query directory {}: {}", OSTREE_DEPLS_PATH, e))?;
     let ostree_depls_data_mtime = FileTime::from_last_modification_time(&ostree_depls_data);
 
     if let Some(cache) = &client.status_cache {
         if cache.mtime == ostree_depls_data_mtime {
-            trace!("cache fresh, using cached rpm-ostree status");
+            trace!("status cache is up to date");
             return Ok(cache.status.clone());
         }
     }
 
     STATUS_CACHE_MISSES.inc();
     trace!("cache stale, invoking rpm-ostree to retrieve local deployments");
-    let status = invoke_cli_status(false)?;
+    let status = Rc::new(invoke_cli_status(false)?);
     client.status_cache = Some(StatusCache {
         status: status.clone(),
```

Comment:
Note that while there's still a `clone()` call here that boils down to (inlined) an integer increment, not a full `memcpy()` of the status data

Reply:
Would this be more obvious if we used `Rc::clone(&status)`?

Reply:
https://rust-lang.github.io/rust-clippy/master/#clone_on_ref_ptr (not enabled by default because it has too many false positives)

Reply:
I see the value in the lint; updated to use `Rc::clone`.